Task9/src/main/java/task2/model/ModelLogic.java

```diff
@@ -0,0 +1,237 @@
+        return sentences.stream()
+                .map(s -> s.replaceAll("[,.?!()]", ""))
+                .flatMap(Pattern.compile(" ")::splitAsStream)
+                .map(s -> s.toLowerCase())
+                .filter(s -> !s.isEmpty())
+                .filter(s -> !s.matches("-?\\d+(\\.\\d+)?"))
+                .filter(s -> !s.matches("\\b\\d+(st|nd|rd|th)\\b"))
+                .distinct()
+                .sorted(compWordsByAlphabet)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<String> sortWordsByPercentageOfVowels() {
+        VowelPercentageComparator vowelPercentageComparator = new VowelPercentageComparator();
+        return sentences.stream()
+                .map(s -> s.replaceAll("[,.?!()]", ""))
+                .flatMap(Pattern.compile(" ")::splitAsStream)
+                .map(s -> s.toLowerCase())
+                .filter(s -> !s.isEmpty())
+                .filter(s -> !s.matches("-?\\d+(\\.\\d+)?"))
+                .filter(s -> !s.matches("\\b\\d+(st|nd|rd|th)\\b"))
+                .distinct()
+                .sorted(vowelPercentageComparator)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<String> sortWordsThatStartWithVowel() {
+        Comparator<String> compWordsByAlphabet = (s1, s2) -> s1.compareTo(s2);
+        Pattern pattern = Pattern.compile("^[AEIOUYaeiouy]");
+        return sentences.stream()
+                .map(s -> s.replaceAll("[,.?!()]", ""))
+                .flatMap(Pattern.compile(" ")::splitAsStream)
+                .filter(s -> pattern.matcher(s).find())
+                .map(s -> s.toLowerCase())
+                .distinct()
+                .sorted(compWordsByAlphabet)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<String> sortAllWordsByNumbersOfGivingLetter(String enterLetter) {
+        return sentences.stream()
+                .map(s -> s.replaceAll("[,.?!()]", ""))
+                .flatMap(Pattern.compile(" ")::splitAsStream)
+                .map(s -> s.toLowerCase())
+                .filter(s -> !s.matches("\\b\\d+(st|nd|rd|th)\\b"))
+                .distinct()
+                .filter(s -> s.contains(enterLetter))
+                .sorted((word1, word2) -> {
+                    long count1 = word1.chars().filter(ch -> ch == enterLetter.charAt(0)).count();
+                    long count2 = word2.chars().filter(ch -> ch == enterLetter.charAt(0)).count();
+                    return (int)(count2 - count1);
+                })
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Map <String, Long> findNumberOfOccursForAllGivenWords(List<String> listOfEnterWords) {
+        return sentences.stream()
+                .map(s -> s.replaceAll("[,.?!()]", ""))
+                .flatMap(Pattern.compile(" ")::splitAsStream)
+                .map(s -> s.toLowerCase())
+                .filter(s -> !s.isEmpty())
```

Comment:
We can create separate method for this repeating lines. Something like "splitSentence" or "getWords"